src/NuGet.Clients/NuGet.PackageManagement.UI/Options/CheckedListBoxAccessibleObject.cs

```diff
@@ -31,7 +32,16 @@ public override AccessibleObject GetChild(int index)
         {
             if (index >= 0 && index < CheckedListBox.Items.Count)
             {
-                var packageSource = (PackageSourceContextInfo)CheckedListBox.Items[index];
+                var item = (PackageSourceContextInfo)CheckedListBox.Items[index];
+                PackageSource packageSource = new PackageSource(item.Source, item.Name);
+                if (packageSource.IsHttp && !packageSource.IsHttps)
+                {
+                    var sourceMessage = string.Concat(
+                        "Warning: Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source."
+                        packageSource.Source);
+                    return new CheckedListBoxItemAccessibleObject(this, packageSource.Name, index, sourceMessage);
+                }
+
```

Comment:
@donnie-msft here I'm adding the warning text, the `CheckedListBoxItemAccessibleObject(CheckedListBoxAccessibleObject parent, string name, int index, string helpText)` will be announced as "name helpText" in Narrator

Reply:
I didn't hear this with Narrator or NVDA. Narrator was reading the expected message.
Actually, NVDA isn't announcing anything for some reason. Have you tested with it?